src/server.cc

```diff
@@ -341,6 +341,7 @@ InferenceServer::Stop(const bool force)
       }
     } else {
       const auto& live_models = model_repository_manager_->LiveModelStates();
+      size_t bg_models_size = model_repository_manager_->BackgroundModelsSize();
```

Comment:
It would probably be good to amend the log output just below as well for clarity on shutdown.

"Found X live models, **Y unloading models**, and Z inflight non-inference requests"

Otherwise, in the event of 0 live and 0 inflight but 1 unloading, the logs could be confusing:
```
Timeout 30: Found 0 live and 0 inflight
Timeout 29: Found 0 live and 0 inflight
Timeout 28: Found 0 live and 0 inflight
...
Timeout failed...
```

Reply:
Following up on https://github.com/triton-inference-server/server/pull/6835#discussion_r1468262293 :

A live model could also be unloading.

How about
"Found X live models, **Y ghost models**, and Z inflight non-inference requests"
or
"Found **(X + Y)** live models and Z inflight"?

I slightly prefer the latter one, because I don't know if there is any test depends on that timeout statement.

Reply:
I think it's fine to sum them like the latter if the background model is enough to block the shutdown in the same way as live models

Reply:
Updated: [Show bg model size when printing timeout info](https://github.com/triton-inference-server/core/pull/323/commits/5aef7f15f3a2fdd400b5c675131771387b16fc1e)
```
Signal (15) received.
I0127 01:32:27.144715 128 server.cc:307] Waiting for in-flight requests to complete.
I0127 01:32:27.144726 128 model_lifecycle.cc:223] StopAllModels()
I0127 01:32:27.144735 128 model_lifecycle.cc:241] InflightStatus()
I0127 01:32:27.144743 128 server.cc:323] Timeout 30: Found 0 model versions that have in-flight inferences
I0127 01:32:27.144750 128 model_lifecycle.cc:390] AsyncUnload() 'identity_fp32'
I0127 01:32:27.144792 128 server.cc:338] All models are stopped, unloading models
I0127 01:32:27.144799 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:27.144808 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:27.144815 128 server.cc:347] Timeout 30: Found 2 live models and 0 in-flight non-inference requests
I0127 01:32:27.144820 128 backend_model_instance.cc:795] Stopping backend thread for identity_fp32_0...
I0127 01:32:27.144822 128 server.cc:353] identity_fp32 v1: UNLOADING
I0127 01:32:27.144861 128 identity.cc:730] TRITONBACKEND_ModelInstanceFinalize: delete instance state
I0127 01:32:27.145098 128 identity.cc:638] TRITONBACKEND_ModelFinalize: delete model state
I0127 01:32:27.145113 128 model_lifecycle.cc:618] OnDestroy callback() 'identity_fp32' version 1
I0127 01:32:27.145121 128 model_lifecycle.cc:620] successfully unloaded 'identity_fp32' version 1
I0127 01:32:28.145006 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:28.145051 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:28.145060 128 server.cc:347] Timeout 29: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:29.145238 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:29.145299 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:29.145313 128 server.cc:347] Timeout 28: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:30.145516 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:30.145572 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:30.145586 128 server.cc:347] Timeout 27: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:31.145719 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:31.145763 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:31.145771 128 server.cc:347] Timeout 26: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:32.146226 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:32.146276 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:32.146285 128 server.cc:347] Timeout 25: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:33.146966 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:33.147048 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:33.147099 128 server.cc:347] Timeout 24: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:34.147361 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:34.147424 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:34.147438 128 server.cc:347] Timeout 23: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:35.148017 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:35.148101 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:35.148116 128 server.cc:347] Timeout 22: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:36.149025 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:36.149092 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:36.149106 128 server.cc:347] Timeout 21: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:37.149693 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:37.149754 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:37.149768 128 server.cc:347] Timeout 20: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:38.150620 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:38.150680 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:38.150694 128 server.cc:347] Timeout 19: Found 1 live models and 0 in-flight non-inference requests
I0127 01:32:38.188567 128 python_be.cc:2342] TRITONBACKEND_ModelFinalize: delete model state
I0127 01:32:38.188650 128 model_lifecycle.cc:618] OnDestroy callback() 'identity_fp32' version 1
I0127 01:32:38.188665 128 model_lifecycle.cc:620] successfully unloaded 'identity_fp32' version 1
I0127 01:32:39.151129 128 model_lifecycle.cc:190] LiveModelStates()
I0127 01:32:39.151212 128 model_lifecycle.cc:265] BackgroundModelsSize()
I0127 01:32:39.151340 128 server.cc:347] Timeout 18: Found 0 live models and 0 in-flight non-inference requests
...
```